base/linalg/dense.jl

```diff
@@ -362,6 +362,36 @@ function inv{T}(A::StridedMatrix{T})
     return convert(typeof(parent(Ai)), Ai)
 end
 
+"""
+    factorize(A)
```

Comment:
It would be great to explain more clearly that all the elements are checked and the structure is determined from this check, in contrast to structure imposed by a type. I'm not sure what the formulation should be but I think it should be mentioned that the algorithm goes through all the values of the matrix.